internal/client/gitea.go

```diff
@@ -0,0 +1,161 @@
+package client
+
+import (
+	"os"
+	"strconv"
+
+	"code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/sdk/gitea"
+	"github.com/apex/log"
+	"github.com/goreleaser/goreleaser/internal/artifact"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	"github.com/goreleaser/goreleaser/pkg/context"
+)
+
+type giteaClient struct {
+	client *gitea.Client
+}
+
+// NewGitea returns a gitea client implementation
+func NewGitea(ctx *context.Context) (Client, error) {
+	token := ctx.Token
+	client := gitea.NewClient(ctx.Config.Release.Gitea.InstanceURL, token)
+	return &giteaClient{client: client}, nil
+}
+
+// CreateFile creates a file in the repository at a given path
+// or updates the file if it exists
+func (c *giteaClient) CreateFile(
+	ctx *context.Context,
+	commitAuthor config.CommitAuthor,
+	repo config.Repo,
+	content []byte,
+	path,
+	message string,
+) error {
+	// doesn't have to be implemented for Gitea
```

Comment:
We would need this method for `brew` and `scoop`. I just checked the sdk has only has a `GetFile` method, whereas the API offers CRUD. Filed an issue at gitea: https://gitea.com/gitea/go-sdk/issues/187

Reply:
So we should wait until they introduce the CRUD features into the SDK?

Reply:
Yes and then make another PR for brew and scoop. Could you add that those 2 are not supported yet for gitea please

Reply:
Sure, I will